docs/suspensive.org/src/lib/hooks/useFetchContributors.ts

```diff
@@ -41,17 +41,19 @@ const useFetchContributors = () => {
           throw new Error('Failed to fetch contributors')
         }
         const data: GithubRepoContributor[] = (await response.json()) as GithubRepoContributor[]
-        const filteredContributors = data.filter((contributor) => {
-          const login = contributor.author.login
-          return !['github-actions[bot]', 'dependabot[bot]', 'renovate[bot]'].includes(login)
-        })
-        setResult({
-          data: filteredContributors,
-          error: undefined,
-          isError: false,
-          isLoading: false,
-          isSuccess: true,
-        })
+        if (Array.isArray(data) && data.length) {
+          const filteredContributors = data.filter((contributor) => {
+            const login = contributor.author.login
+            return !['github-actions[bot]', 'dependabot[bot]', 'renovate[bot]'].includes(login)
+          })
+          setResult({
+            data: filteredContributors,
+            error: undefined,
+            isError: false,
+            isLoading: false,
+            isSuccess: true,
+          })
+        }
```

Comment:
Even if the data is not an array or the length is 0, the fetch success status must be expressed as state. Please modify isSuccess to be true so that the correct server status state can be displayed even if the length of the data is 0 or there is no data when the fetch is successful.

Reply:
thank you, I've just pushed the change
It will keep `isSuccess` true for all 200 OK, even though the response data is empty or non-array
plus, I refactored the code for better exception messages

Reply:
Thanks!